app/routes/groups_.new.tsx

```diff
@@ -27,29 +28,64 @@ export async function action({ request }: ActionFunctionArgs) {
     return { error: { message: `Form not submitted correctly.` } };
   }
 
-  let groupImageUrl = null;
+  const formObject = Object.fromEntries(form);
+  const newGroupFrom = await z
+    .object({
+      groupName: z.string().min(1, 'Group name is required.'),
+      description: z.string().min(1, 'Description is required.'),
+      groupImage: z
+        .instanceof(File)
+        .refine(
+          (file) => !file || file?.size <= MAX_FILE_SIZE_MB * 1024 * 1024,
+          `File size can't exceed ${MAX_FILE_SIZE_MB}MB.`,
+        )
+        .refine(
+          (file) => ACCEPTED_IMAGE_TYPES.split(', ').includes(file.type),
+          `Only ${ACCEPTED_IMAGE_TYPES} are supported.`,
+        ),
+    })
+    .safeParseAsync(formObject);
+
+  if (!newGroupFrom.success) {
+    return badRequest({
+      success: false,
+      fields: {
+        groupName: formObject.groupName,
+        description: formObject.description,
+        groupImage: formObject.groupImage,
+      },
+      error: { message: `the following fields contains errors: ${newGroupFrom.error}` },
+    });
+  }
+
+  let groupImageUrl = '';
   if (groupImage && groupImage instanceof File && groupImage.name) {
     // Create a single supabase client for interacting with your database
     if (!process.env.SUPABASE_SECRET) throw new Error('SUPABASE_SECRET is not defined');
     const supabase = createClient('https://fzzehiiwadkmbvpouotf.supabase.co', process.env.SUPABASE_SECRET);
     const uuid = crypto.randomUUID();
-    const { data, error } = await supabase.storage
-      .from('group-cover-images')
-      .upload(`${uuid}-${groupImage.name}`, groupImage, {
-        cacheControl: '3600',
-        upsert: false,
-      });
-    if (error) {
-      return { error: { message: `Error uploading image: ${error.message}` } };
+    try {
+      const { data, error } = await supabase.storage
+        .from('group-cover-images')
+        .upload(`${uuid}-${groupImage.name}`, groupImage, {
+          cacheControl: '3600',
+          upsert: false,
+        });
+      groupImageUrl = `https://fzzehiiwadkmbvpouotf.supabase.co/storage/v1/object/public/group-cover-images/${data?.path}`;
+      if (error) {
+        return { error: { message: `Error uploading image: ${error.message}` } };
+      }
+    } catch (error) {
+      console.error('Error uploading image:', error);
```

Comment:
Should we return an error here as well? What would be the pros & cons of either approach?